src/workerd/jsg/util.h

```diff
@@ -294,7 +294,7 @@ struct LiftKj_ {
     auto isolate = info.GetIsolate();
     try {
       try {
-        v8::HandleScope scope(isolate);
+        KJ_DASSERT(v8::HandleScope::NumberOfHandles(isolate) > 0);
```

Comment:
@kentonv @danlapid this should address your concerns.

Reply:
What does this prove?

Reply:
This isn't a useful check: it's obviously always true, because we have to have at least one HandleScope in order to run JS code at all, and we were called by JS, so obviously there is a HandleScope somewhere.

What we need to know is that the immediate caller has already created a HandleScope that wraps just this call.

Your other comment does seem to show at least some call sites allocate a HandleScope which will be destroyed immediately upon return from the API call, though it's hard to see whether all call sites are covered. It's frustrating that V8 doesn't document a guarantee...